src/main/java/com/greenfox/hackathon/controller/WishlistController.java

```diff
@@ -0,0 +1,45 @@
+package com.greenfox.hackathon.controller;
+
+import com.greenfox.hackathon.exception.NoSuchItemException;
+import com.greenfox.hackathon.exception.UserDoesNotExistException;
+import com.greenfox.hackathon.model.Item;
+import com.greenfox.hackathon.model.WishlistDTO;
+import com.greenfox.hackathon.service.WishlistService;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Controller;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PutMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+
+import java.security.Principal;
+
+@Controller
+public class WishlistController {
+  private WishlistService wishlistService;
+
+  @Autowired
+  public WishlistController(WishlistService wishlistService) {
+    this.wishlistService = wishlistService;
+  }
+
+  @GetMapping("/wishlist")
+  public ResponseEntity<WishlistDTO> getWishlistByUser(Principal principal) throws UserDoesNotExistException {
+    return ResponseEntity.ok(wishlistService.getWishlist(principal));
+  }
+
+  @PutMapping("/wishlist")
+  public ResponseEntity<Void> addItem(Principal principal, @RequestBody Item item) throws UserDoesNotExistException {
```

Comment:
Instead of an item it would be better with just an itemId. From frontend viewpoint.